connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointConfig.java

```diff
@@ -166,6 +173,29 @@ Duration consumerPollTimeout() {
         return Duration.ofMillis(getLong(CONSUMER_POLL_TIMEOUT_MILLIS));
     }
 
+    public List<ConfigValue> validate() {
+        List<ConfigValue> invalidConfigs = new ArrayList<>();
+
+        if (!this.getBoolean(EMIT_CHECKPOINTS_ENABLED) && !this.getBoolean(SYNC_GROUP_OFFSETS_ENABLED)) {
+            Arrays.asList(new ConfigValue(SYNC_GROUP_OFFSETS_ENABLED), new ConfigValue(EMIT_CHECKPOINTS_ENABLED))
+                    .forEach(configValue -> {
+                        configValue.addErrorMessage("MirrorCheckpointConnector can't run without both " + SYNC_GROUP_OFFSETS_ENABLED + ", " +
+                                EMIT_CHECKPOINTS_ENABLED + " set to false");
+                        invalidConfigs.add(configValue);
+                    });
+        }
+
+        if ("false".equals(Optional.ofNullable(this.originals().get(EMIT_OFFSET_SYNCS_ENABLED)).orElse("true"))) {
+            Arrays.asList(new ConfigValue(SYNC_GROUP_OFFSETS_ENABLED), new ConfigValue(EMIT_CHECKPOINTS_ENABLED), new ConfigValue(EMIT_OFFSET_SYNCS_ENABLED))
```

Comment:
I can see a rationale for reporting errors on all of these properties, but 1) IMO it's overkill and a bit redundant, and 2) it's also misleading if either offset syncing or checkpointing are disabled.

Can we just attach the error to the `emit.offset-syncs.enabled` property?

Reply:
refactored this now